src/client/executor.rs

```diff
@@ -530,82 +523,79 @@ impl Client {
         let start_time = Instant::now();
         let command_result = match connection.send_raw_command(raw_cmd, request_id).await {
             Ok(response) => {
-                match T::Response::deserialize_response(&response) {
```

Comment:
Because `#[serde(flatten)]` doesn't play nicely with deserializing raw BSON values (see https://github.com/serde-rs/serde/issues/2106), response deserialization was changed back to being handled in the individual `handle_response` implementations of the various operations like it was before the performance improvements work. However, this time we can use `RawDocument` to extract the various things we need here that are not operation specific (e.g. operationTime) without having to deserialize into `Document` first, which was a big bottleneck before.

This also helps enable the `Cursor/ChangeStream::with_type` API being discussed in #518.